brave/src/main/java/brave/propagation/ExtraField.java

```diff
@@ -0,0 +1,330 @@
+ * on the wire as "baggage-country-code" and "baggage-user-id" respectively.
+ *
+ * <pre>{@code
+ * requestId = ExtraField.create("x-vcap-request-id");
+ * userId = ExtraField.newBuilder("user-id").prefix("baggage-").build();
+ * countryCode = ExtraField.newBuilder("country-code").prefix("baggage-").build();
+ *
+ * // Later, you can call below to affect the country code of the current trace context
+ * ExtraField.setValue("country-code", "FO");
+ * String countryCode = ExtraField.getValue("country-code");
+ * }</pre>
+ *
+ * <h3>Appropriate usage</h3>
+ * It is generally not a good idea to use the tracing system for application logic or critical code
+ * such as security context propagation.
+ *
+ * <p>Brave is an infrastructure library: you will create lock-in if you expose its apis into
+ * business code. Prefer exposing your own types for utility functions that use this class as this
+ * will insulate you from lock-in.
+ *
+ * <p>While it may seem convenient, do not use this for security context propagation as it was not
+ * designed for this use case. For example, anything placed in here can be accessed by any code in
+ * the same classloader!
+ *
+ * @see ExtraFieldPropagation
+ * @see CorrelationFieldScopeDecorator
+ */
+public class ExtraField {
+  /**
+   * Creates a field that is referenced the same in-process as it is on the wire. For example, the
+   * name "x-vcap-request-id" would be set as-is including the prefix.
+   *
+   * @param name will be implicitly lower-cased
+   * @see #name()
+   * @since 5.11
+   */
+  public static ExtraField create(String name) {
+    return new Builder(name).build();
+  }
+
+  /**
+   * Creates a builder for the specified {@linkplain #name()}.
+   *
+   * @param name will be implicitly lower-cased
+   * @see #name()
+   * @since 5.11
+   */
+  public static Builder newBuilder(String name) {
+    return new Builder(name);
+  }
+
+  /**
+   * Returns the value of the field with the specified key or null if not available.
+   *
+   * <p>Prefer using {@link ExtraField#getValue(TraceContext)} when you have a reference to the
+   * underlying field.
+   */
+  @Nullable public static String getValue(TraceContext context, String name) {
+    return PropagationFields.get(context, validateName(name), ExtraFieldPropagation.Extra.class);
+  }
+
+  /**
+   * Returns the value of the field with the specified key or null if not available.
+   *
+   * <p>Prefer using {@link #getValue()} when you have a reference to the underlying field.
+   * <p>Prefer {@link #getValue(TraceContext, String)} if you have a reference to the trace
+   * context.
+   *
+   * @see ExtraField#name()
+   */
+  @Nullable public static String getValue(String name) {
+    TraceContext context = currentTraceContext();
+    return context != null ? getValue(context, name) : null;
+  }
+
+  /**
+   * Sets the value of the field with the specified key, or drops if not a configured field.
```

Comment:
`or drops if not a configured field` what do you mean it's just a no-op call like ignored ? `or ignored if the key does not reference a configured field` perhaps

Reply:
thx